subsys/net/l2/ethernet/arp.h

```diff
@@ -54,10 +54,8 @@ struct arp_entry {
 	uint32_t req_start;
 	struct net_if *iface;
 	struct in_addr ip;
-	union {
-		struct net_pkt *pending;
-		struct net_eth_addr eth;
-	};
+	struct net_eth_addr eth;
+	struct k_fifo pending_queue;
```

Comment:
Couldn't those still be in a union?

Reply:
They can, yes. As part of a future update, I think it makes sense to rewrite this entirely, an ARP entry should only be storing the information it needs and an ARP request should only be storing the information it needs i.e. 2 different lists rather than combining them into 1.

Reply:
Good point